covenant-signer/README.md

```diff
@@ -0,0 +1,335 @@
+    type: local
+
+```
+
+Here, the `key` field contains the base64-encoded public key. After decoding, 
+this key:
+
+```shell
+023a79b546c79d7f7c5ff20620d914b5cf7250631d12f6e26427ed9d3f98c5ccb1
+```
+
+Matches the public key derived earlier and seen in the outputs of `getaddressinfo` 
+and `derive-child-key`.
+
+Congratulations! You have successfully imported your keys from the prior setup 
+and verified your setup for the covenant emulator.
+
+## 5. Create the configuration file
+
+Use the example configuration [file](./example/config.toml) to create your own 
+configuration file. Then, replace the placeholder values with your own 
+configuration.
+
+```toml
+[keystore]
+keystore-type = "cosmos"
+
+[keystore.cosmos]
+# pointing to the directory where the key is stored, unless specified otherwise
+key-directory = "/path/to/keydir"
+
+# the backend to be used for storing the key, in this case file
+keyring-backend = "file"
+
+# the name of the key you used when importing the key
+key-name = "your-key-name"
+
+# the chain id of the chain to be used
+chain-id = "your-chain-id"
+
+[server-config]
+# The IP address where the covenant-signer server will listen
+host = "127.0.0.1"
+# The TCP port number where the covenant-signer server will listen
+port = 9791
+
+[metrics]
+# The IP address where the Prometheus metrics server will listen
+host = "127.0.0.1"
+# This port is used to expose metrics that can be scraped by Prometheus
+port = 2113
+```
+
+Parameters:
+- `keystore-type`: Type of keystore used, e.g., "cosmos" for Cosmos SDK compatibility.
+- `key-directory`: Path where keys are stored on the filesystem.
+- `keyring-backend`: Backend system for key management, e.g., "file", "os".
+- `key-name`: Name of the key used for signing transactions.
+- `chain-id`: Identifier of the blockchain network.
+- `host` (server-config): IP address where the server listens, typically "127.0.0.1" for local access.
+- `port` (server-config): TCP port number for the server.
+- `host` (metrics): IP address for the Prometheus metrics server, typically "127.0.0.1".
+- `port` (metrics): TCP port number for the Prometheus metrics server.
+
+## 6. Running the Covenant Signer
+
+The covenant signer can be run using the following command:
+
+```shell
+covenant-signer start --config ./path/to/config.toml
+```
+
+The covenant signer must be run in a secure network and only accessible by the 
+covenant emulator.
+
+Once the covenant signer is set up and unlocked, you can configure the covenant 
+emulator to use it. The URL of the covenant signer, which is 
+http://127.0.0.1:9791, should be specified in the covenant emulator's 
+configuration file under the `remotesigner` section.
+
+It's important to note that the key specified in the covenant emulator's 
+configuration is not the covenant key itself. Instead, it is a 
+key used for sending Cosmos transactions.
+
+## 7. Using the covenant signer for signing transactions
```

Comment:
I think this guide mixes transition and operational concepts together without separating them.

I recommend that this document is split in the following sections:

1. Pre-requisites
2. Installation
3. Transitioning your covenant key from the phase-1 setup
4. Operation
    a. Config
    b. Starting the daemon
    c. Unlocking the key
    d. Testing the setup